app/components/task-card.js

```diff
@@ -1,51 +1,108 @@
 import EmberCan from 'ember-can';
 import Ember from 'ember';
+import createTaskUserOptions from 'code-corps-ember/utils/create-task-user-options';
 
 const {
   Component,
   computed,
   computed: { alias, mapBy },
   get,
-  inject: { service }
+  getProperties,
+  inject: { service },
+  isEmpty
 } = Ember;
 
 export default Component.extend({
   attributeBindings: ['data-can-reposition', 'data-model-id', 'data-model-type'],
   classNames: ['task-card'],
-  classNameBindings: ['canReposition:task-card--can-reposition', 'isLoading:task-card--is-loading', 'taskTypeClass'],
+  classNameBindings: ['canReposition:task-card--can-reposition', 'isLoading:task-card--is-loading'],
   tagName: 'div',
 
+  currentUser: service(),
   session: service(),
+  store: service(),
+  taskAssignment: service(),
 
-  trueValue: true,
+  bound: false,
+  shouldShowUsers: false,
 
   ability: EmberCan.computed.ability('task'),
   canAssign: alias('ability.canAssign'),
   canEdit: alias('ability.canEdit'),
   canReposition: alias('ability.canReposition'),
 
+  currentUserId: alias('currentUser.user.id'),
+
   'data-can-reposition': computed('canReposition', function() {
     let canReposition = get(this, 'canReposition');
     return canReposition ? 'true' : 'false';
   }),
   'data-model-id': alias('task.id'),
   'data-model-type': 'task',
 
-  assignedUser: alias('task.assignedUser'),
-
   isLoading: alias('task.isLoading'),
 
-  taskType: alias('task.taskType'),
-  taskTypeClass: computed('taskType', function() {
-    return `task-card--${get(this, 'taskType')}`;
+  selectedOption: computed('taskUser', 'users', function() {
+    let user = get(this, 'taskUser');
+    let users = get(this, 'users');
+    if (isEmpty(user) || isEmpty(users)) {
+      return;
+    }
+    return users.find((item) => {
+      return get(user, 'id') === get(item, 'id');
+    });
   }),
+
   taskSkills: mapBy('task.taskSkills', 'skill'),
+  taskUserId: alias('taskUser.id'),
+  users: alias('members'),
+
+  userOptions: computed('currentUser.user', 'taskUserId', 'users', function() {
+    let { currentUserId, taskUserId, users } = getProperties(this, 'currentUserId', 'taskUserId', 'users');
```

Comment:
Any specific reason why the computed depends on `currentUser.user`. but then uses `currentUserId`? Does it not work if it depends on `currentUserId`?